ipinfo/client.go

```diff
@@ -30,20 +30,36 @@ type Client struct {
 	// Cache interface implementation to prevent API quota overuse for
 	// identical requests.
 	Cache *Cache
+
+	// The API token used for authorization for more data and higher limits.
+	Token string
 }
 
-// NewClient returns a new IPinfo API client. If a nil httpClient is provided,
-// http.DefaultClient will be used. To use authenticated API methods provide
-// http.Client with AuthTransport.
-func NewClient(httpClient *http.Client) *Client {
+// NewClient returns a new IPinfo API client.
+//
+// If `httpClient` is nil, `http.DefaultClient` will be used.
+//
+// If `cache` is nil, no cache is automatically assigned. You may set one later
+// at any time with `client.SetCache`.
+//
+// If `token` is empty, the API will be queried without any token. You may set
+// one later at any time with `client.SetToken`.
+func NewClient(
```

Comment:
I don't like using `cache` and `token` arguments here. It leads to ugly code like the following to initialize API client with the default HTTP client, no cache, and no token:

```
client := ipinfo.NewClient(nil, nil, "")
```